core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2800,11 +2803,19 @@ object SparkContext extends Logging {
       // large enough if any task resources were specified.
       taskResourceRequirements.foreach { taskReq =>
         val execAmount = executorResourcesAndAmounts(taskReq.resourceName)
-        if (taskReq.amount * numSlots < execAmount) {
+        if ((numSlots * taskReq.amount / taskReq.numParts) < execAmount) {
+          val taskReqStr = if (taskReq.numParts > 1) {
+            s"${taskReq.amount}/${taskReq.numParts}"
```

Comment:
In this case the `taskReq.amount` should always be `1` ?

Reply:
Yes that is correct. Though I am not sure this hurts. I like the idea of seeing it in text if it is misconfigured, but let me know if you want that changed.